src/openmc_heat_driver.cpp

```diff
@@ -214,42 +368,73 @@ void OpenmcHeatDriver::set_heat_source()
   }
 }
 
+void OpenmcHeatDriver::update_density()
+{
+  if (this->has_global_coupling_data()) {
+    std::copy(densities_.begin(), densities_.end(), densities_prev_.begin());
+  }
+
+  densities_ = heat_driver_->density();
+  int solid_offset = heat_driver_->n_pins_ * heat_driver_->n_rings() * heat_driver_->n_axial_;
```

Comment:
This is a narrowing conversion, since the operands are size types.  Maybe declare `solid_offset` as `auto`?